firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,212 @@
+      THEN("The data in the output_crcelement's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with a complete 4-byte header, and a payload consistent with the CRC field of the header, is given to the receiver")
+      THEN("the transform method reports ok status")
+      THEN("After the transform method is called, the value returned by the output_crcelement's crc accessor method is equal to the crc field of the input_buffer body's header")
+      THEN("The payload buffer returned by the output_crcelement's payload accessor method is equal to the payload from the input_buffer body")
+      THEN("The data in the output_crcelement's payload buffer is independent of the data in input_buffer")
+
+  GIVEN("A CRC element receiver of capacity 254 bytes, and output_crcelement constructed with an non-empty payload buffer and non-zero crc member")
+    // assertions carried out in GIVEN:
+    // The value returned by the output_crcelement's crc accessor method is '0x40D06D79'
+    // The payload buffer returned by the output_crcelement's payload accessor method is '0x23 0x84 0xf5 0x6a 0xac 0xae'
+
+    WHEN("An empty input_buffer is given to the crc element receiver")
+      THEN("the transform method reports invalid parse status")
+      THEN("After the transform method is called, the value returned by the output_crcelement's crc accessor method remains unchanged")
+      THEN("The output_crcelement's payload accessor method returns a reference to the payload given in its constructor")
+      THEN("The data in the output_crcelement's payload buffer is independent of the data in input_buffer")
+
+    WHEN("A body with a complete 4-byte header, and a payload consistent with the CRC field of the header, is given to the receiver")
+      THEN("the transform method reports ok status")
+      THEN("After the transform method is called, the value returned by the output_crcelement's crc accessor method is equal to the crc field of the input_buffer body's header")
+      THEN("The payload buffer returned by the output_crcelement's payload accessor method is equal to the payload from the input_buffer body")
+      THEN("The data in the output_crcelement's payload buffer is independent of the data in input_buffer")      
+
+/* CRCElementSender
+Scenario: CRCElementSender: correctly generates CRCElement bodies from payloads
+
+  GIVEN("A CRC element sender of capacity 254 bytes")
+
+    WHEN("A payload is given along with a sufficiently large output buffer")
+      THEN("the transform method reports ok status")
+      THEN("output buffer is as expected, equal to (crc + input_buffer)")
+      THEN("The CRC field of the output buffer's header is consistent with the input_payload.")
+
+  GIVEN("A CRC element sender of buffer size 20 bytes")
+
+    WHEN("The output buffer is not large enough for the input payload")
+      THEN("the transform status is equal to invalid_length")
```

Comment:
What should the contents of output_buffer be after the transform method is called? (I think the contents should be unchanged, whether output_buffer is empty or not)